tests/bluetooth/host/keys/bt_keys_get_addr/src/test_suite_full_list_no_overwrite.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include "testing_common_defs.h"
+#include <addr.h>
+#include <keys.h>
+#include "mocks/hci_core_expects.h"
+
+/*
+ *  Test adding extra (ID, Address) pair while the keys pool list is full
+ *
+ *  Constraints:
+ *   - Keys pool list is full
+ *   - CONFIG_BT_KEYS_OVERWRITE_OLDEST isn't enabled
+ *
+ *  Expected behaviour:
+ *   - NULL reference pointer is returned
+ */
+void test_adding_new_pair_to_full_list(void)
+{
+	struct bt_keys *returned_key;
+	uint8_t id = BT_ADDR_ID_3;
+	bt_addr_le_t *addr = BT_ADDR_LE_3;
+
+	returned_key = bt_keys_get_addr(id, addr);
+
+	expect_not_called_bt_unpair();
+
+	zassert_true(returned_key == NULL, "bt_keys_get_addr() returned a non-NULL reference");
+}
+
+static bool pragma_overwrite_not_enabled(const void *s)
```

Comment:
I'd name that one `overwrite_oldest_disabled`. Pragma is misleading because it means something in the context of the preprocessor. If you want something, you could use `predicate`, that's what the function signature defines.

Reply:
Your point is totally valid, I just followed the name used in the test framework getting started guide, but it's misleading.
I changed it to `predicate` as used in the current version of the guide.